include/mls/credential.h

```diff
@@ -38,6 +39,20 @@ struct BasicCredential
   static const CredentialType type;
 };
 
+struct X509Credential
+{
+  X509Credential() {}
+
+  explicit X509Credential(const std::vector<bytes>& der_chain);
```

Comment:
You can offer this if you want, but what you really want (especially on deserialize) is `X509Credential(const bytes& der_chain)`.  Unless we're going to change the spec to have a separate `opaque` per cert.  Which might be a good idea.